### ISSUE TITLE
Help setting group labels and different, matches and different bars on a special setup 

### ISSUE THREAD

**Issue by ronniedroid** [qtile/qtile#1528]
Hello

I started using two monitors with qtile and I finally managed to make it the way I like (different groups, pinned to different screens) 

The problem is, I don't know how to set different bars for the two monitors, and can't set labels to groups and can't set matches (rules) 

The implementation I have (copied from another issue posted here a while back) is

```
pinned_groups = ['123', '456']
all_groups = ''.join(pinned_groups)

groups = [Group(i) for i in all_groups]

for j, names in enumerate(pinned_groups):
    keys.extend(Key([mod], i, lazy.to_screen(j), lazy.group[i].toscreen()) for i in names)

keys.extend(Key([mod, 'shift'], i, lazy.window.togroup(i)) for i in all_groups)

screens = [Screen(top=bar.Bar([widget.GroupBox(visible_groups=list(names))], 30))
           for names in pinned_groups]
```

Hope you guys can help :-)

**Comment by ramnes** [comment]
I guess that the issue here is more of a Python issue, as `bar.Bar([widget.GroupBox(visible_groups=list(names))` is evaluated only once and given to all your screens. Try instantiating different bars and then assigning them to each screen.

**Comment by pawamoy** [comment]
Why would `bar.Bar([widget.GroupBox(visible_groups=list(names))` be evaluated only once? In `[f(n) for n in names]`, `f(n)` is evaluated at each iteration.

So here, at each iteration, `Screen(top=bar.Bar([widget.GroupBox(visible_groups=list(names))], 30))` should be a new screen with a new bat with a new list containing a new group box widget, no?

**Comment by ramnes** [comment]
You're perfectly right, not sure what I meant or thought, but it definitely looks like I misread the code! That being said, without an answer from OP, this is going nowhere, so I'll go ahead and close. Feel free to comment if this is still an issue!
